pkg/cmd/pulumi/cmd/project_sdk_link.go

```diff
@@ -0,0 +1,816 @@
+// project, in the absence of us attempting to perform this linking automatically.
+func linkPythonPackage(ws pkgWorkspace.Context, root string, pkg *schema.Package, out string) error {
+	fmt.Printf("Successfully generated a Python SDK for the %s package at %s\n", pkg.Name, out)
+	fmt.Println()
+	proj, _, err := ws.ReadProject()
+	if err != nil {
+		return err
+	}
+	packageSpecifier, err := filepath.Rel(root, out)
+	if err != nil {
+		return err
+	}
+
+	modifyRequirements := func() error {
+		f, err := os.OpenFile(
+			filepath.Join(root, "requirements.txt"),
+			os.O_CREATE|os.O_APPEND|os.O_RDWR,
+			0o600,
+		)
+		if err != nil {
+			return fmt.Errorf("error opening requirments.txt: %w", err)
+		}
+		defer f.Close()
+
+		_, err = f.WriteString(packageSpecifier + "\n")
+		if err != nil {
+			return fmt.Errorf("error appending to requirments: %w", err)
+		}
+
+		cmd := exec.Command("pulumi", "install")
+		cmd.Stderr = os.Stderr
+		cmd.Stdout = os.Stdout
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("error running %s: %w", cmd.String(), err)
+		}
+
+		return nil
+	}
+	options := proj.Runtime.Options()
+	if toolchain, ok := options["toolchain"]; ok {
+		if tc, ok := toolchain.(string); ok {
+			var depAddCmd *exec.Cmd
+			switch tc {
+			case "pip":
+				if err := modifyRequirements(); err != nil {
+					return err
+				}
+			case "poetry":
+				depAddCmd = exec.Command("poetry", "add", packageSpecifier)
+			case "uv":
+				depAddCmd = exec.Command("uv", "add", packageSpecifier)
+			default:
+				return fmt.Errorf("unsupported package manager: %s", tc)
+			}
+
+			if depAddCmd != nil {
+				depAddCmd.Stderr = os.Stderr
+				depAddCmd.Stdout = os.Stdout
+				err = depAddCmd.Run()
+				if err != nil {
+					return fmt.Errorf("error running %s: %w", depAddCmd.String(), err)
+				}
+			}
+		} else {
+			return fmt.Errorf("packagemanager option must be a string: %v", toolchain)
+		}
+	} else {
+		// Assume pip if no packagemanager is specified
+		if err := modifyRequirements(); err != nil {
+			return err
+		}
+	}
+
+	pyInfo, ok := pkg.Language["python"].(python.PackageInfo)
+	var importName string
+	if ok && pyInfo.PackageName != "" {
+		importName = pyInfo.PackageName
+	} else {
+		importName = strings.ReplaceAll(pkg.Name, "-", "_")
+	}
+
+	fmt.Println()
+	fmt.Println("You can then import the SDK in your Python code with:")
+	fmt.Println()
+	fmt.Printf("  import pulumi_%s as %s\n", importName, importName)
+	fmt.Println()
+	return nil
+}
+
+// Prints instructions for linking a locally generated SDK to an existing Go
+// project, in the absence of us attempting to perform this linking automatically.
+func linkGoPackage(root string, pkg *schema.Package, out string) error {
+	fmt.Printf("Successfully generated a Go SDK for the %s package at %s\n", pkg.Name, out)
+
+	// All go code is placed under a relative package root so it is nested one
+	// more directory deep equal to the package name.  This extra path is equal
```

Comment:
Feel like this could be worded differently. One directory deeper? Equal was not what I was expecting. Might just be me though.